clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/AddExecutor.scala

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.clio.client.dispatch
+
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.headers.HttpCredentials
+import io.circe.{Decoder, Encoder}
+import io.circe.parser.parse
+import org.broadinstitute.clio.client.commands.{AddCommand, ClioCommand}
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClioWebClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AddExecutor(addCommand: AddCommand) extends Executor {
+  override def execute(webClient: ClioWebClient, ioUtil: IoUtil)(
+    implicit ec: ExecutionContext,
+    credentials: HttpCredentials
+  ): Future[HttpResponse] = {
+
+    val location = addCommand.metadataLocation
+    val index = addCommand.index
+
+    implicit val decoder: Decoder[index.metadataType] = index.decoder
+    implicit val encoder: Encoder[index.metadataType] = index.encoder
+
+    val parsedOrError = parse(IoUtil.readMetadata(location)).left.map { err =>
+      new RuntimeException(
+        s"Could not parse contents of $location as JSON.",
+        err
+      )
+    }
+
+    val decodedOrError = parsedOrError
+      .flatMap(_.as[index.metadataType])
+      .left
+      .map { err =>
+        new RuntimeException(
+          s"Invalid metadata given at $location. Run the '${ClioCommand.getSchemaPrefix}${index.commandName}' command to see the expected JSON format for ${index.name}s.",
```

Comment:
Should there be a `-` between `${ClioCommand.getSchemaPrefix}` and `${index.commandName}`?

Reply:
No, the prefixes end in `-`